Sources/Amplitude/Utilities/EventPipeline.swift

```diff
@@ -54,32 +54,36 @@ public class EventPipeline {
         eventCount = 0
         guard let storage = self.storage else { return }
         storage.rollover()
-        guard let eventFiles: [URL]? = storage.read(key: StorageKey.EVENTS) else { return }
-        cleanupUploads()
-        if pendingUploads == 0 {
-            for eventFile in eventFiles! {
-                guard let eventsString = storage.getEventsString(eventBlock: eventFile) else {
-                    continue
+        guard let eventFiles: [URL] = storage.read(key: StorageKey.EVENTS) else { return }
+        for eventFile in eventFiles {
+            uploadsQueue.sync {
```

Comment:
We are now uploading one file at a time?

Reply:
we are starting one upload at a time, as of now we are unbounded as to the number of simultaneous uploads.